### PR TITLE
[Enhancement] Use the number of hardware threads as default sink dop

### PR DESCRIPTION
**Pull request by letian-jiang**
Why I'm doing:

What I'm doing:

Use the number of hardware threads as default sink dop.

If `pipeline_sink_dop` is set, use it as the sink dop. Otherwise, use the number of hardware and cap it to 64. 

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exec/pipeline/fragment_executor.cpp [hunk 1]
```diff
@@ -1050,7 +1055,7 @@ Status FragmentExecutor::_decompose_data_sink_to_operator(RuntimeState* runtime_
                 fragment_ctx);
 
         size_t source_dop = fragment_ctx->pipelines().back()->source_operator_factory()->degree_of_parallelism();
-        size_t sink_dop = request.pipeline_sink_dop();
+        size_t sink_dop = _calc_sink_dop(ExecEnv::GetInstance(), request);
```
Comment (kevincai) [inline be/src/exec/pipeline/fragment_executor.cpp:1058]: May consider the function returns uint32_t. It is an implicit overflow/underflow if returns a negative int32_t and assigned to size_t.

Comment (letian-jiang) [reply be/src/exec/pipeline/fragment_executor.cpp:1058]: The implementation ensures it outputs a positive number.

Comment (kevincai) [reply be/src/exec/pipeline/fragment_executor.cpp:1058]: so why not `uint32_t _calc_sink_dop()` if it can only returns a positive number?

Comment (letian-jiang) [reply be/src/exec/pipeline/fragment_executor.cpp:1058]: ok.

**Review by dirtysalt — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Pull request merged**